tests/unit/libutil-support/tests/string_callback.hh

```diff
@@ -0,0 +1,12 @@
+#pragma once
+#include <string>
+
+namespace nix::testing {
+
+void observe_string_cb(const char * start, unsigned int n, std::string * user_data);
+inline void * observe_string_cb_data(std::string & out) {
+    return (void *) &out;
+};
+#define OBSERVE_STRING(str) (void *)nix::testing::observe_string_cb, nix::testing::observe_string_cb_data(str)
+
+}
```

Comment:
```suggestion
trailing newline? :)
```

Reply:
I thought editorconfig did something, but now I've added to vscode

```json
  "[c]": {
    "files.insertFinalNewline": true
  },
  "[cpp]": {
    "files.insertFinalNewline": true
  },
```